### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by mzottola**

### CODE REVIEW

File: src/main/resources/db/migration/V1.2__insert_admin_user.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO user(username, password, role, version)
```
Comment (jeromelebaron) [inline src/main/resources/db/migration/V1.2__insert_admin_user.sql]: Where do you specifiy this patch for liquibase?

Comment (mzottola) [reply src/main/resources/db/migration/V1.2__insert_admin_user.sql]: This is not liquibase, I use flyway to manage database migrations: see the commit message of first comit https://github.com/mzottola/crossfit-registration/commit/5d04c107ef14893b6cd8a614b75b2357b85da017
I use the default configuration, as explained [in the official doc](https://flywaydb.org/documentation/migrations#discovery-1)

Comment (jeromelebaron) [reply src/main/resources/db/migration/V1.2__insert_admin_user.sql]: 👍 I didn't know that, I will read it

File: app/src/main/resources/db/migration/V1.1__modify_user_table.sql [hunk 2]
```diff
@@ -1,4 +1 @@
 ALTER TABLE user ADD COLUMN role VARCHAR(100) NOT NULL;
-
-INSERT INTO user(username, password, role, version)
-VALUES('admin', 'admin', 'ADMIN', 0);
```
Comment (jeromelebaron) [inline app/src/main/resources/db/migration/V1.1__modify_user_table.sql]: Are you sure you can do that? If this patch was already applied on database, the checksum changed and flyway will block you. You need to do another one with this change no? However it's not in production yet, so maybe it's not important now.

Comment (mzottola) [reply app/src/main/resources/db/migration/V1.1__modify_user_table.sql]: > However it's not in production yet

Exactly.When the first release will be done, I will autosquash every file into 1 migration file.

File: app/src/main/java/com/crossfit/security/auth/UserAuthenticationServiceImpl.java [hunk 3]
```diff
@@ -8,27 +8,36 @@
...
-    public UserAuthenticationServiceImpl(final UserRepository userRepository) {
+    public UserAuthenticationServiceImpl(final UserRepository userRepository,
+                                         final PasswordEncoder passwordEncoder) {
         this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
         this.tokenByUsers = new HashMap<>();
     }
 
     @Override
     public Optional<String> login(final String username, final String password) {
```
Comment (joeyrogues) [inline app/src/main/java/com/crossfit/security/auth/UserAuthenticationServiceImpl.java]: Si dans le future on passe en serverless:lambda on aura cette logique dans un autre service/function

Comment (mzottola) [reply app/src/main/java/com/crossfit/security/auth/UserAuthenticationServiceImpl.java]: Yes, c'est pour ça que je l'ai isolé dans un Service :nerd_face:

**Pull request merged**